### PR TITLE
Add square wasi without gitignore

### PR DESCRIPTION
**Pull request by calvinrp**
This is a branch off of https://github.com/Lay3rLabs/avs-toolkit/pull/23 that can replace that PR.

Closes https://github.com/Lay3rLabs/avs-toolkit/issues/16

The [instructions to configure your local setup and build the component](https://github.com/Lay3rLabs/avs-toolkit/tree/add-square-wasi-without-gitignore/wasi/square) will get simpler before the hackathon. You shouldn't need to manually edit a file, it will be a CLI command or something configurable in a local toml file (Cargo.toml or another file).

### CODE REVIEW

File: wasi/square/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+If you have the [cargo-binstall](https://github.com/cargo-bins/cargo-binstall)
+utility installed, `cargo component` can also be installed via a prebuilt
+release artifact, saving time on the installation:
+
+```
+cargo binstall cargo-component
+```
+
+The configuration for registry mappings is in the process of getting better,
```
Comment (ethanfrey) [inline wasi/square/README.md:38]: Thanks for these docs, I never would have figured it out on my own.

**Review by ethanfrey — APPROVED** [review]
Going to merge this now as it is a big step forward.
I think there is more to do here for DevX.

One small comment (for you) is that if we commit `bindings.rs` and it causes `cargo fmt` issues, this is a bit annoying. When does it get rebuilt? 

Is it possible to add a `rustfmt.toml` file such that `bindings.rs` would be correctly formatted? (I am not too particular on formating styles.. just odd that auto-gen fails on cargo fmt and nice if this "just works")

File: wasi/square/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+The default location is `$XDG_CONFIG_HOME/wasm-pkg/config.toml` on unix-like systems and
+`{FOLDERID_RoamingAppData}\wasm-pkg\config.toml` on Windows. Examples of this are found below:
+
+| Platform | Path                                            |
+| -------- | ----------------------------------------------- |
+| Linux    | `/home/<username>/.config`                      |
+| macOS    | `/Users/<username>/Library/Application Support` |
+| Windows  | `C:\Users\<username>\AppData\Roaming`           |
+
+The configuration file is TOML and currently must be edited manually. A future release will include
```
[the remark this answers is not in the document]
Comment (calvinrp) [reply wasi/square/README.md:50]: `cargo-component` shares the `wkg` config. Installing `wkg` is optional for this workflow. This is all very new. Days old and actively being improved.

Yep, I will document. Put in Wasmatic repo?

**Comment by calvinrp** [comment]
> One small comment (for you) is that if we commit `bindings.rs` and it causes `cargo fmt` issues, this is a bit annoying. When does it get rebuilt?

It gets rebuilt with the `cargo component build` and `cargo component publish` commands. Not with the standard `cargo build` etc commands.


> Is it possible to add a `rustfmt.toml` file such that `bindings.rs` would be correctly formatted? (I am not too particular on formating styles.. just odd that auto-gen fails on cargo fmt and nice if this "just works")

Created an [issue](https://github.com/bytecodealliance/cargo-component/issues/340) in cargo component.

**Comment by calvinrp** [comment]
> Going to merge this now as it is a big step forward. I think there is more to do here for DevX.

💯 Agree that the DX needs a lot more work. But fixable and can improve incrementally every day. Expect to be in a better place before and of this week.

**Pull request merged**